bundles/org.openhab.binding.automower/src/main/java/org/openhab/binding/automower/internal/rest/api/automowerconnect/AutomowerConnectApi.java

```diff
@@ -0,0 +1,141 @@
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.openhab.binding.automower.internal.rest.api.HusqvarnaApi;
+import org.openhab.binding.automower.internal.rest.api.automowerconnect.dto.MowerCommandRequest;
+import org.openhab.binding.automower.internal.rest.api.automowerconnect.dto.MowerListResult;
+import org.openhab.binding.automower.internal.rest.api.automowerconnect.dto.MowerResult;
+import org.openhab.binding.automower.internal.rest.exceptions.AutomowerCommunicationException;
+import org.openhab.binding.automower.internal.rest.exceptions.UnauthorizedException;
+
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * Allows access to the AutomowerConnectApi
+ *
+ * @author Markus Pfleger - Initial contribution
+ */
+@NonNullByDefault
+public class AutomowerConnectApi extends HusqvarnaApi {
+
+    public AutomowerConnectApi(HttpClient httpClient) {
+        super(httpClient);
+    }
+
+    @Override
+    public String getBaseUrl() {
+        return "https://api.amc.husqvarna.dev/v1";
+    }
+
+    public MowerListResult getMowers(String appKey, String token) throws AutomowerCommunicationException {
+        final Request request = getHttpClient().newRequest(getBaseUrl() + "/mowers");
+        request.method(HttpMethod.GET);
+
+        request.header("Authorization-Provider", "husqvarna");
+        request.header("Authorization", "Bearer " + token);
+        request.header("X-Api-Key", appKey);
+        request.header("Content-Type", "application/vnd.api+json");
+
+        ContentResponse response;
+        try {
+            response = request.send();
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            throw new AutomowerCommunicationException(e);
+        }
+
+        return parseResponse(response, MowerListResult.class);
+    }
+
+    public MowerResult getMower(String appKey, String token, String mowerId) throws AutomowerCommunicationException {
+        final Request request = getHttpClient().newRequest(getBaseUrl() + "/mowers/" + mowerId);
+        request.method(HttpMethod.GET);
+
+        request.header("Authorization-Provider", "husqvarna");
+        request.header("Authorization", "Bearer " + token);
+        request.header("X-Api-Key", appKey);
+        request.header("Content-Type", "application/vnd.api+json");
+
+        ContentResponse response;
+        try {
+            response = request.send();
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            throw new AutomowerCommunicationException(e);
+        }
+
+        return parseResponse(response, MowerResult.class);
+    }
+
+    public void sendCommand(String appKey, String token, String id, MowerCommandRequest command)
+            throws AutomowerCommunicationException {
+        final Request request = getHttpClient().newRequest(getBaseUrl() + "/mowers/" + id + "/actions");
```

Comment:
You need to add a timeout to the request. This can be set in the method I described in the other comment so you don't have to set it only in 1 place.

Reply:
Ok, I thought there would be a proper default timeout. I will use 10 seconds...